yt/data_objects/construction_data_containers.py

```diff
@@ -683,6 +685,7 @@ def __init__(
             )
             + self.ds.domain_offset
         )
+        self._extra_data_source = data_source  # copy instead of reference???
```

Comment:
reference is OK here -- if we change anything, I'd suggest changing the field parameters, but even that's not necessary.  I think I made a stink about this in-person but that was misguided.

Can I namesmith it to `_base_data_source` or something?

Reply:
ya! name change is good

Reply:
The real problem is with the internal usage of `data_source`, not anything you did -- but I don't want this to get too far out of scope.  I think what you have is good.

Reply:
name changed to `_base_data_source`

Reply:
out of date now that it's being stored in _data_source... will resolve.